frostsnapp/lib/wallet.dart

```diff
@@ -716,3 +846,107 @@ Uri getBlockExplorer(BitcoinNetwork network) {
     return Uri.parse("https://mempool.space/signet/");
   }
 }
+
+class BannerStateProvider extends InheritedWidget {
+  final bool showBanner;
+
+  const BannerStateProvider({
+    required this.showBanner,
+    required Widget child,
+    Key? key,
+  }) : super(key: key, child: child);
+
+  static BannerStateProvider? of(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<BannerStateProvider>();
+  }
+
+  @override
+  bool updateShouldNotify(BannerStateProvider oldWidget) {
+    return showBanner != oldWidget.showBanner;
+  }
+}
+
+class BackupWarningBanner extends StatelessWidget {
+  final FrostKey frostKey;
+  final BackupManager backupManager;
+
+  const BackupWarningBanner({
+    required this.frostKey,
+    required this.backupManager,
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final backupManager = FrostsnapContext.of(context)!.backupManager;
+    final walletCtx = WalletContext.of(context)!;
+    final backupStream = walletCtx.backupStream;
+
+    final bannerState = BannerStateProvider.of(context);
+    // Don't show if we're in new key flow and haven't dismissed dialog
+    if (bannerState != null && !bannerState.showBanner) {
+      return const SliverToBoxAdapter(child: SizedBox.shrink());
```

Comment:
Just don't construct `BackupWarningBanner` if showBanner is false.

Reply:
If I remember correctly this was to prevent:
1) Clicking begin backup,
2) Immediately backing out
3) User can't see banner, when we probably do want to show it (so they don't go and receive a million bitcoins right away)

Though your design sounds better, maybe see if you can avoid that behavior?